src/shared/lib/timers/AppTimer.ts

```diff
@@ -0,0 +1,62 @@
+import TimerBase from './TimerBase';
+
+class AppTimer extends TimerBase {
+  private duration: number;
+  private startTime?: number;
+  private onFinish: Function;
+
+  constructor(onFinish: Function, startImmediately: boolean, duration: number) {
+    super(startImmediately);
+    this.duration = duration;
+    this.onFinish = onFinish;
+  }
+
+  start(): void {
+    super.start();
+    this.startTime = Date.now();
+    this.setTimer();
+  }
+
+  stop() {
+    super.stop();
+    clearTimeout(this.timerId);
+  }
+
+  setTimer(duration = this.duration) {
+    this.timerId = setTimeout(() => {
+      this.onFinish();
+    }, duration);
+  }
+
+  private isDurationOver(): boolean {
+    const timerTimeDiff = Date.now() - this.startTime!;
+    return !!(this.duration && timerTimeDiff > this.duration);
+  }
+
+  private geTimeLeftAterBackground() {
```

Comment:
Typo in the name. Please install https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker